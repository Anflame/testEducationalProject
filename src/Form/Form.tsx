import logo from "./logo.svg";
import { useState, useEffect } from "react";

function Form() {
  const [data, setData] = useState<null | object>(null);
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState("");

  const onClick = () => setToggle((prev) => !prev);

  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 100);
  });

  return (
    <div>
      <h1 data-testid="value-heading">{value}</h1>
      {toggle && <div data-testid="toggle">toggle</div>}
      {data && <div>data</div>}
      {/* data */}
      <input
        style={{ color: "red" }}
        type="text"
        placeholder="input value1"
        value={value}
        onChange={(e) => setValue(e.target.value)}
      />
      <button data-testid="toggle-btn" onClick={onClick} type="submit">
        button
      </button>
      Hello React
    </div>
  );
}

export default Form;
