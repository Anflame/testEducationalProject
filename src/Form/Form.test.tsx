import { fireEvent, render, screen } from "@testing-library/react";
import Form from "./Form";

describe("Test App", () => {
  test("renders learn react link", async () => {
    render(<Form />);

    const linkElement = screen.getByText(/hello react/i);
    expect(linkElement).toBeInTheDocument();

    const btn = screen.getByRole("button");
    expect(btn).toBeInTheDocument();

    const input = screen.getByPlaceholderText(/input value/i);
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
    expect(input).toHaveStyle({ color: "red" });

    screen.debug();

    const dataEl = await screen.findByText(/data/i);
    expect(dataEl).toBeInTheDocument();

    const helloWorldEl = screen.queryByText(/hello world/i);
    expect(helloWorldEl).toBe(null);
  });

  test("click event", () => {
    render(<Form />);
    const btn = screen.getByTestId("toggle-btn");

    expect(screen.queryByTestId("toggle")).toBeNull();
    fireEvent.click(btn);
    expect(screen.queryByTestId("toggle")).toBeInTheDocument();
    fireEvent.click(btn);
    expect(screen.queryByTestId("toggle")).toBeNull();
  });
  test("input event", () => {
    render(<Form />);
    const input = screen.getByPlaceholderText(/input value/i);

    expect(screen.queryByTestId("value-heading")).toContainHTML("");
    fireEvent.input(input, {
      target: {
        value: "1234",
      },
    });
    expect(screen.queryByTestId("value-heading")).toContainHTML("1234");
  });
});
