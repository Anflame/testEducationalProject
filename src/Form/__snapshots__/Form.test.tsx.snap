// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App renders learn react link 1`] = `
<input
  placeholder="input value1"
  style="color: red;"
  type="text"
  value=""
/>
`;
